### ISSUE TITLE
API for loaded pipeline config?

### ISSUE THREAD

**Issue by ricpruss** [automl/Auto-PyTorch#8]
I love how Auto-Pytorch is coming along but it is hard to see how a pipeline config is being interpreted before fitting the pipeline.
For example if I am setting a non-default loss in a config file.  I can do a:
selector = autonet.pipeline[LossModuleSelector.get_name()]
selected_loss = autonet.pipeline[selector.get_name()].fit_output['loss_function']
and then I can examine that to get to know that my config did what was expected but before running the pipeline something like:
autonet.pipeline.get_pipeline_config()
Shows all possible loss functions not the one selected in the config file.
Sorry if I am missing something obvious,
Ric

**Comment by mlindauer** [comment]
Dear Ric,

Thank you for your interest in Auto-PyTorch.

@LMZimmer Could you please look into it and try to help Ric.

Best,
Marius

**Comment by LMZimmer** [comment]
Hey Ric,

we're happy about your interest in Auto-PyTorch.

If you want to know the config used during training use `autonet.get_current_autonet.config()`.

Using the `pipeline.get_pipeline_config()` method will always give the default config as it is not modified during training (unless you initialize the autonet with a pipeline). Rather the config that `autonet.get_current_autonet.config()` returns is passed to the optimization algorithm on the basis of which it then optimizes.

I hope this helps. If not, or if u you have any other issues, feel free to contact us.

Best,
Lucas

**Comment by ricpruss** [comment]
Thanks that works. The structure of pipelines, nodes (which are a parallel linear sequence) and the encompassing autonets configured by defaults, and loadable text configs and limited by conditions is powerful but complex.  Using it sort of feels like directed graphs on a blackboard design pattern thanks for publishing it,
Ric
